Edit the Enterprise migration guide

ptgott opened the pull request:
**Edit the guide for structure and style:**
   - Use title case in the title.
   - Wrap lines at 80 chars.
   - Begin H2s with "Step n/d"
   - Move the Conclusion text to the intro. How-to guides do not include conclusions in our docs, and this information is valuable at the start of the guide.
   - Use numbered steps for instructions.
   - Use relative file paths for links.
   - Minor style edits.

**Remove the "Verify connectivity" step:** This repeats instructions from the step to deploy a cloud-hosted cluster.

**Add a note on migrating plugins and Machine ID Bots:** the guide only includes instructions for migrating agents in the section on migrating services. This change suggests linking to docs on migrating plugins and Bots.

**Edit the navigation:**
  - Add a link to the navigation sidebar.
  - Edit the "Choose a Teleport Edition" intro page to (a) imply that cloud-hosted is the recommended solution and (b) link to the migration guide.

ptgott committed:
Edit the Enterprise migration guide

**Edit the guide for structure and style:**
   - Use title case in the title.
   - Wrap lines at 80 chars.
   - Begin H2s with "Step n/d"
   - Move the Conclusion text to the intro. How-to guides do not include
     conclusions in our docs, and this information is valuable at the start
     of the guide.
   - Use numbered steps for instructions.
   - Use relative file paths for links.
   - Minor style edits.

**Remove the "Verify connectivity" step:** This repeats instructions
from the step to deploy a cloud-hosted cluster.

**Add a note on migrating plugins and Machine ID Bots:** the guide only
includes instructions for migrating agents in the section on migrating
services. This change suggests linking to docs on migrating plugins and
Bots.

**Edit the navigation:**
  - Add a link to the navigation sidebar.
  - Edit the "Choose a Teleport Edition" intro page to (a) imply that
    cloud-hosted is the recommended solution and (b) link to the
    migration guide.

webvictim commented on docs/pages/choose-an-edition/migrate-to-cloud.mdx:
```diff
@@ -0,0 +1,324 @@
+h1: Migrate to Cloud-Hosted Teleport Enterprise 
+---
+
+Migrating from a Teleport Enterprise (self-hosted) deployment to Teleport
+Enterprise (cloud-hosted) offers scalability, reliability, and ease of
+management. By following the steps outlined in this guide, you can transition
+your Teleport Enterprise (self-hosted) deployment to Teleport Enterprise
+(cloud-hosted) successfully while ensuring minimal disruption to your
+operations.
```
I'm probably yelling into the void and the decision has already been made above my pay grade, but I really don't like this whole "same product name with different qualifier in brackets" style. It's just clunky and awkward to read.

webvictim commented on docs/pages/choose-an-edition/migrate-to-cloud.mdx:
````diff
@@ -0,0 +1,324 @@
+cluster. You should be able to connect to both Teleport clusters and execute
+`tctl` commands using your current credentials.
+
+1. Log in to the self-hosted Teleport Enterprise cluster:
+
+   ```code
+   $ tsh login --proxy=<Var name="enterprise.example.com" /> --user=<Var name="myuser" />
+   $ tctl status
+   ```
+
+1. Log in to the cloud-hosted Teleport Enterprise cluster:
+
+   ```code
+   $ tsh login --proxy=<Var name="example.teleport.sh" /> --user=<Var name="myuser" />
+   $ tctl status
+   ```
````
Might be worth adding SSO examples here, `--auth okta` or whatever (where `--user` will be ignored)

webvictim commented on docs/pages/choose-an-edition/migrate-to-cloud.mdx:
```diff
@@ -0,0 +1,324 @@
+  title="Note"
+>
+
+  Trusted clusters are not supported on Teleport Enterprise (cloud-hosted). You
+  will not be able to migrate trusted cluster resources.
```
It might be worth mentioning this much closer to the top, so people don't get this far and then realise it won't work for them.

webvictim commented on docs/pages/choose-an-edition/migrate-to-cloud.mdx:
```diff
@@ -0,0 +1,324 @@
+
+</Admonition>
+
+If you are using infrastructure-as-code tools to 
+To achieve this:
```
This seems to be a little misjoined

webvictim commented on docs/pages/choose-an-edition/migrate-to-cloud.mdx:
```diff
@@ -0,0 +1,324 @@
+
+1. Stop Teleport services on the agent (if applicable).
+
+1. Update the `proxy_server` or `auth_servers` field in the agent configuration
```
Might be worth just mentioning `proxy_server` here, with a note saying "If you don't have an `auth_server` or `auth_servers` entry rather than a `proxy_server` entry, you should migrate your config to `version: v3` and switch to using `proxy_server` as in the example)

Folks might get confused between `auth_server` (string used in config version `v3`) and `auth_servers` (list used in config version `v1/v2`) and break connectivity.

webvictim commented on docs/pages/choose-an-edition/migrate-to-cloud.mdx:
````diff
@@ -0,0 +1,324 @@
+   rm -rf /var/lib/teleport
+   ```
+
+1. On Linux servers, restartthe Teleport process on each agent or, if you are
````
```suggestion
1. On Linux servers, restart the Teleport process on each agent or, if you are

```

ptgott committed:
Respond to webvictim feedback

- Reduce the overuse of `Teleport Enterprise (cloud|self-hosted)`.
- Move the note re: trusted clusters to the Prerequisites.
- Mention the `--auth` flag in `tsh login` commands.
- Fix typo.
- Add a note about the `proxy_server` field

oshati reviewed (commented):
Looks good to me!
